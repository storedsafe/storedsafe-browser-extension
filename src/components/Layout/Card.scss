@import '../../variables';

.card {
  padding: $spacing;
  background-color: #fff;
  border-radius: $border-radius;
  box-shadow: 0 1px 5px 0 rgba(0, 0, 0, 0.1);
}
