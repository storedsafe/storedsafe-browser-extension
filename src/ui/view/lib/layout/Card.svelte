<script lang="ts" module>
  import type { Snippet } from "svelte";

  export interface Props {
    /** Background css color */
    backgroundColor?: string;
    color?: string;
    children?: Snippet;
  }
</script>

<script lang="ts">
  let {
    backgroundColor = "var(--color-bg)",
    color = "var(--color-fg)",
    children,
  }: Props = $props();
</script>

<!--
  @component
  Card-style container for creating uniform page content.
-->
<article
  class="grid card"
  style={`
    --card-bg: ${backgroundColor};
    --card-fg: ${color};
    `}
>
  {@render children?.()}
</article>

<style>
  article {
    overflow: hidden auto;
    background-color: var(--card-bg);
    color: var(--card-fg);
    padding: var(--spacing);
    box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.2);
    border-radius: var(--border-radius);
  }

  .card :global(h1),
  .card :global(h2),
  .card :global(h3),
  .card :global(h4),
  .card :global(h5),
  .card :global(h6) {
    color: var(--color-primary);
    text-align: left;
  }
</style>
