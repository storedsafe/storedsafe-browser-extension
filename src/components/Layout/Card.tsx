import * as React from 'react';
import './Card.scss';

export interface CardProps {
  className?: string;
  children?: React.ReactNode;
}

export const Card: React.FC<CardProps> = ({
  children,
  className = '',
}: CardProps) => <section className={`card ${className}`}>{children}</section>
