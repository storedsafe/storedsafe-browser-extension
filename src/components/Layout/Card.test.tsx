import * as React from 'react';
import { shallow } from 'enzyme';
import { Card } from './Card';

test('<Card />', () => {
  const wrapper = shallow(<Card />);
  expect(wrapper).toMatchSnapshot();
});
test('<Card children />', () => {
  const wrapper = shallow(<Card>Hello</Card>);
  expect(wrapper).toMatchSnapshot();
});
