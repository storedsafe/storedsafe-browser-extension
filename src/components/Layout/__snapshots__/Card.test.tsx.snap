// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> 1`] = `ShallowWrapper {}`;

exports[`<Card children /> 1`] = `ShallowWrapper {}`;
